### DISCUSSION TITLE
Digit Consistency Check does not work with dig-class11

### DISCUSSION THREAD

**Discussion by friedpa** [jomjol/AI-on-the-edge-device#2343] [General]
### The Problem

Obwohl Negative Rates nicht erlaubt sind, wird trotzdem ein kleiner Wert als vorher akzeptiert. Interessanterweise wird bei der Rolling Version ab und zu zum Raw Value 0,1 dazugezählt und als Value ausgegeben (inkl. dem setzen des Previous Value Wertes).

![grafik](https://user-images.githubusercontent.com/42890695/223710200-e7889ea7-0814-403a-b60b-b97b8611ba59.png)
![grafik](https://user-images.githubusercontent.com/42890695/223710281-285c1eff-d656-4536-95f1-f1ccf1dcc5e4.png)

### Version

Development-Branch: rolling (Commit: 1f5d4de+)

### Logfile

```shell
No Logfile
```

**Comment by friedpa** [comment]
Heute wieder:
![grafik](https://user-images.githubusercontent.com/42890695/224015877-ed843005-0845-4809-b2ee-3f4c24eacb96.png)
eigentlich müsste da doch ein "negative Rate" Error ausgegeben werden, oder?

Edit:
Kann der Fehler mit "Check Digit Increase Consistency" zusammenhängen, der war früher immer Flase gestzt und jetzt steht er nach 15.0.3 auf true. Habe diesen Parameter nun wieder auf false gesetzt und werde berichten.

**Comment by ghost** [comment]
Welches Neuronales Netz verwendest du?

**Comment by ghost** [comment]
Testfall mit den Werten oben erstellt => kein Fehler.
Man sieht auch, dass der Value den Pre-Value behält. Nur RawValue enthält den gelesenen Wert.
Aber "error" sollte nicht auf "no error" stehen.

@friedpa : Mich würde noch der Parameter ExtendedResolution interessieren neben der Modelart.

**Comment by friedpa** [comment]
Extended Resolution = false
dig0640s3.ftlite => alt aber hat immer sehr gut funktioniert

**Comment by friedpa** [comment]
Es muss mit dem "Check Digit Increase Consistency" zusammenhängen, da nach setzen auf false der Fehler (bisher) nicht mehr auftritt.

**Comment by ghost** [comment]
Danke. Also dig-class11, dann macht auch Check Digit Increase Consistency Sinn als mögliche Quelle.
Ich konnte es nachstellen. Es hat hier gar nichts mit Allow negatives zu tun, sondern mit der Sonderbehandlung Check Digit Consistency, die den Wert verändert.

Hier wird der Pre-Value-Wert aus dem vorherigen Cycle mit dem aktuellen Wert verglichen.
Um dem Fehler auf die Spur zu kommen, müsste ich auch den PreValue Wert (bzw. Value davor) kennen. Viellicht kannst du beim nächsten Mal das Log mitschicken (am besten so lange das Log auf debug stellen).

**Comment by caco3** [comment]
I am changing it to a discussion

**Comment by caco3** [comment]
@friedpa If you see it again, fell free to comment again for further diagnosis.
